test/string_view-test/string_view-test.cpp

```diff
@@ -0,0 +1,349 @@
+/*
+* string_view-test.cpp
+* Ray Chen, Sean Murthy 
+* (c) 2020 sigcpp https://sigcpp.github.io. See LICENSE.MD
+*
+* Attribution and copyright notice must be retained.
+* - Attribution may be augmented to include additional authors
+* - Copyright notice cannot be altered
+* Attribution and copyright info may be relocated but they must be conspicuous.
+*
+* Test the string_view of the C++ standard library and 
+* provide a test benchmark for subsequent implementation of string_view.
+* - see C++17 [string.view]
+*	https://timsong-cpp.github.io/cppwp/n4659/string.view
+*/
+
+#include <stdexcept>
+
+#include "../verifiers.h"
+
+void string_view_test()
+{
+	using std::string_view;
+
+	//1. Construction and assignment
+	//2. Capacity
+
+	// Default constructor. Constructs an empty string_view. 
+	string_view empty_sv;
+	is_zero(empty_sv.size(), "empty_sv.size()");
+	is_true(empty_sv.length() == empty_sv.size(), "empty_sv.length() == empty_sv.size()");
+	is_true(empty_sv.empty(), "empty_sv.empty()");
+
+	//constructs a string_view with const char[]
+	const char c_hw[]{ "hello, world!" };
+	string_view sv(c_hw);
+	is_true(sv.size() == 13, "sv.size()");
+	is_true(sv.length() == sv.size(), "sv.length() == sv.size()");
+	is_false(sv.empty(), "!sv.empty()");
+
+	//Constructs a view of the first 5 characters of the character array
+	string_view f_sv(c_hw, 5);
```

Comment:
I suggest using a non-C-string char array for this test case.

Reply:
add a non-C-string char array for this test case. 2830432